src/Tuura/Function.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE TypeFamilies #-}
+module Tuura.Function (
+    printBooleanFunction
+    ) where
+
+import Tuura.Graph
+
+-- data Function = Open String | Closed String deriving (Eq, Ord)
+
+instance Graph Function where
+    type Vertex Function = String
+    empty       = Closed ""
+    vertex      = Closed . id
+    overlay p q = Open $ printFunction p ++ " * " ++ printFunction q
+    connect p _ = Closed $ printFunction p
+
+foldFunction :: b -> (String -> b) -> (b -> b -> b) -> (b -> b -> b) -> String -> GraphExpr String -> b
+foldFunction e v o c s ge = case ge of
+        Empty       -> e
+        Vertex x    -> v x
+        Overlay x y -> if test s x
```

Comment:
Can you try to rewrite this using guards `| test s x = ...` instead of `if` statements? It might look nicer.

Reply:
I'm pretty sure this whole thing can be simplified a lot using the Alga library.

Reply:
Guards won't work inside a case statement. But I can use further case statements instead. This may or may not add to confusion, but wait for the next commit to view, and we can decide.